pkg/component/gardener/dashboard/configmap.go

```diff
@@ -127,6 +127,26 @@ func (g *gardenerDashboard) configMap(ctx context.Context) (*corev1.ConfigMap, e
 			},
 		}
 
+		if g.values.OIDC.CertificateAuthoritySecretRef != nil {
+			caSecret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      g.values.OIDC.CertificateAuthoritySecretRef.Name,
+					Namespace: g.namespace,
+				},
+			}
+
+			if err := g.client.Get(ctx, client.ObjectKeyFromObject(caSecret), caSecret); err != nil {
+				return nil, fmt.Errorf("failed reading referenced ca secret: %w", err)
+			}
+
+			caData, ok := caSecret.Data["ca.crt"]
+			if !ok {
+				return nil, fmt.Errorf("failed reading ca secret: missing ca.crt key")
+			}
+
+			cfg.OIDC.CA = string(caData)
```

Comment:
How about validation that the passed string is a valid certificate? You can use https://github.com/gardener/gardener/blob/db182529c3c1badd987f7e5b20bdcc71f33f8e0a/pkg/utils/encoding.go#L90-L98 function here for that.